src/global-event/index.tsx

```diff
@@ -20,25 +18,30 @@ export interface RegisteredListeners {
 	document: ListenerObject;
 }
 
-export class GlobalEvent extends WidgetBase<GlobalEventProperties> {
-	private _listeners: RegisteredListeners = {
+const factory = create({
+	destroy,
+	diffProperty
+}).properties<GlobalEventProperties>();
+
+export const GlobalEvent = factory(function({ children, middleware: { destroy, diffProperty } }) {
+	const listeners: RegisteredListeners = {
```

Comment:
this needs to be put into an `icache` @ericos  as they won't be saved across renders.

Reply:
Good idea. Updated.